pkg/storage/internalstorage/util.go

```diff
@@ -83,11 +83,20 @@ func applyListOptionsToQuery(query *gorm.DB, opts *pediainternal.ListOptions) *g
 				jsonQuery.NotEqual(requirement.Value)
 			case selection.Equals, selection.DoubleEquals:
 				jsonQuery.Equal(requirement.Value)
+			default:
+				continue
 			}
 			query = query.Where(jsonQuery)
 		}
 	}
 
+	var amount *int64
+	if opts.AttachRemainingCount != nil && *opts.AttachRemainingCount {
+		var count int64
+		amount = &count
+		query = query.Count(amount)
```

Comment:
```suggestion
		query = query.Count(&amount)
```

Reply:
amount is a pointer

Reply:
Yes, pointer should work.

Reply:
``` golang
	var amount = new(int64)
	query = query.Count(amount)
```
Or
``` golang
	var amount int64
	query = query.Count(&amount)
```

Reply:
amount needs to be returned as a pointer, nil and 0 have different meanings